src/server/lock_manager/config.rs

```diff
@@ -3,26 +3,47 @@
 use super::deadlock::Scheduler as DeadlockScheduler;
 use super::waiter_manager::Scheduler as WaiterMgrScheduler;
 use configuration::{rollback_or, ConfigChange, ConfigManager, Configuration, RollbackCollector};
+use serde::de::{Deserialize, Deserializer};
 
 use std::error::Error;
+use tikv_util::config::{rduration_deser_from_str, ReadableDuration};
 
 #[derive(Clone, Serialize, Deserialize, PartialEq, Debug, Configuration)]
 #[serde(default)]
 #[serde(rename_all = "kebab-case")]
 pub struct Config {
     #[config(skip)]
     pub enabled: bool,
-    pub wait_for_lock_timeout: u64,
-    pub wake_up_delay_duration: u64,
+    #[serde(deserialize_with = "compat_desr")]
+    pub wait_for_lock_timeout: ReadableDuration,
+    #[serde(deserialize_with = "compat_desr")]
+    pub wake_up_delay_duration: ReadableDuration,
     pub pipelined: bool,
 }
 
+#[derive(Deserialize)]
+#[serde(untagged)]
+enum StrOrNum<'a> {
```

Comment:
You can use [serde_json::Value](https://docs.serde.rs/serde_json/value/enum.Value.html).